tests/rptest/tests/compatibility/kafka_streams_test.py

```diff
@@ -387,3 +393,208 @@ def __init__(self, test_context):
+        example2.wait()
+
+        # Start the driver and wait for it to load
+        driver.start()
+        driver.wait()
+
+        host = example1.node_name()
+
+        # Do HTTP GET on the possible routes
+        self._http_get(host, "/state/instances",
+                       lambda text: "windowed-word-count" in text)
+
+        self._http_get(host, "/state/instances/word-count",
+                       lambda text: "word-count" in text)
+
+        self._http_get(host, "/state/instance/word-count/hello",
+                       lambda text: "word-count" in text)
+
+        self._http_get(host, "/state/keyvalue/word-count/hello",
+                       lambda text: "hello" in text)
+
+        # HTTP GET on /state/keyvalues/word-count/all often
+        # responds with 500 error.
+
+        driver.stop()
+        example2.stop()
+        example1.stop()
+
+
+# Not using any KafkaStreams bases classes here because this
+# test runs a single process in the background (a.k.a BackgroundThreadService)
+# but the other tests run multiple background processes.
+class KafkaStreamsMicroservices(KafkaStreamsTest):
+    """
+    Test KafkaStreams Microservices examples which demonstrate an order management
+    workflow, such as one might find in retail and online shopping.
+    """
+
+    topics = (
+        TopicSpec(name="orders"),
+        TopicSpec(name="warehouse-inventory"),
+        TopicSpec(name="order-validations"),
+        TopicSpec(name="payments"),
+        TopicSpec(name="customers"),
+        TopicSpec(name="platinum"),
+        TopicSpec(name="gold"),
+        TopicSpec(name="silver"),
+        TopicSpec(name="bronze"),
+    )
+
+    Example = KafkaStreamExamples.KafkaStreamsMicroservices
+
+    def __init__(self, test_context):
+        super(KafkaStreamsMicroservices,
+              self).__init__(test_context=test_context,
+                             enable_pp=False,
+                             enable_sr=False)
+
+    @cluster(num_nodes=4)
+    @matrix(jar=[
+        "EmailServiceTest", "FraudServiceTest", "InventoryServiceTest",
+        "OrderDetailsServiceTest", "OrdersServiceTest",
+        "ValidationsAggregatorServiceTest"
+    ])
+    def test_kafka_streams_microservices(self, jar):
+        example = self.create_example(jar)
+        example.start()
+        example.wait()
+        example.stop()
+
+
+# Not using any of the KafkaStreams base classes because this test
+# runs a client-side script to "reset" the app in the middle of execution
+# that other tests do not use.
+class KafkaStreamsAppReset(KafkaStreamsTest):
+    """
+    Test KafkaStreams ApplicationReset example that uses confluent's cleanUp API
+    to reset an application between runs.
+    """
+    topics = (
+        TopicSpec(name="my-input-topic"),
+        TopicSpec(name="my-output-topic"),
+        TopicSpec(name="rekeyed-topic"),
+    )
+
+    Example = KafkaStreamExamples.KafkaStreamsAppReset
+    PRODUCER = AnomalyProducer
```

Comment:
It's better not to introduce aliases for the classes: it increases the amount of information we need to keep in the head when we read code and make it harder to search through the code. For example, if we want to find all uses of `AnomalyProducer` we need to search for `AnomalyProducer`, then for `PRODUCER`, then to filter out `RegionProducer` etc

Reply:
The alias was a recommendation from a previous PR. 

I can make the change but I'm not sure I agree with your use-case. Searching for all occurrences of `AnomalyProducer` only yields the lines with `PRODUCER = AnomalyProducer` and the class definition. The search results would not have any occurrences of `RegionProducer` or others.